src/Nethermind/Nethermind.State/PersistentStorageProvider.cs

```diff
@@ -501,23 +516,31 @@ public StorageTree Create(Address address, IScopedTrieStore trieStore, Hash256 s
             => new(trieStore, storageRoot, logManager);
     }
 
-    private sealed class SelfDestructDictionary<TValue>(TValue destructedValue)
+    private sealed class SelfDestructDictionary<TValue>(TValue defaultValue)
     {
-        private bool _selfDestruct;
+        private bool _missingAreDefault;
         private readonly Dictionary<UInt256, TValue> _dictionary = new(Comparer.Instance);
 
+        public void MarkEmpty()
+        {
+            _missingAreDefault = true;
+        }
+
```

Comment:
We potentially don't need this method, we will have one additional clear on dictionary but it is a no-op, so no big deal. We also need to rename `SelfDestruct` to `MarkEmpty` or `MarkCleared`.

Reply:
`ClearAndSetMissingAsDefault`